### DISCUSSION TITLE
关于keep-alive的exclude配置组件名不生效

### DISCUSSION THREAD

**Discussion by liuhe-trec** [vuejs/router#1961] [Help and Questions]
我的代码这样:
```
<keep-alive :exclude="['My','Home']">
```
使用的是setup的ts,按照文档说"在 3.2.34 或以上的版本中，使用 <script setup> 的单文件组件会自动根据文件名生成对应的 name 选项，无需再手动声明。"
我使用的是vue 3.3.4. 在我的My.vue没有显示声明,结果是没有生效.
如果加上 defineOptions 
```
<script setup lang="ts">
defineOptions({
  name: 'My'
})
</script>
```
就可以生效了,这么操作规范吗?

**Comment by wtto00** [comment]
https://codesandbox.io/p/sandbox/unruffled-silence-ngt4ws

我测试的，是可以的。看上面的 demo 链接。
> 使用 <script setup> 的单文件组件会自动根据文件名生成对应的 name 选项，无需再手动声明。
